packages/@o3r/schematics/src/utility/migration/migration.ts

```diff
@@ -0,0 +1,60 @@
+import type { LoggerApi } from '@angular-devkit/core/src/logger';
+import { chain, type Rule } from '@angular-devkit/schematics';
+import { intersects, Range, validRange } from 'semver';
+
+/** Create the migration  */
+interface MigrateRuleRunnerOptions {
+  /** The original version from which to execute the rules */
+  from: string;
+
+  /**
+   * The last version for which to execute the rule.
+   * If not specified, all the versions from the "from" parameter will be considered.
+   */
+  to?: string;
+}
+
+/** Mapping of rules to apply to it's specific range */
+export interface MigrationRulesMap {
+  /** Rules to apply to a specific semver range */
+  [range: string]: Rule | Rule[];
```

Comment:
what do you expect here? The range for the minor? Do we expect a specific pattern here?
Do we recommend carret or tilde?

Reply:
A Semver Range (cf https://docs.npmjs.com/cli/v6/using-npm/semver#ranges), not sure it makes sense to provide any recommendation here as it is completely up to the rules and the way we want it to be applied.
(Angular does not recommend anything neither).